apps/src/lib/client/rpc.rs

```diff
@@ -2609,9 +2886,99 @@ fn lookup_alias(ctx: &Context, addr: &Address) -> String {
 }
 
 /// A helper to unwrap client's response. Will shut down process on error.
-fn unwrap_client_response<T>(response: Result<T, queries::tm::Error>) -> T {
+pub(super) fn unwrap_client_response<T>(
+    response: Result<T, queries::tm::Error>,
+) -> T {
     response.unwrap_or_else(|err| {
         eprintln!("Error in the query {}", err);
         cli::safe_exit(1)
     })
 }
+
+/// Look up the denomination of a token in order to format it
+/// correctly as a string.
+pub(super) async fn format_denominated_amount(
+    client: &HttpClient,
+    token: &Address,
+    sub_prefix: &Option<Key>,
+    amount: token::Amount,
+) -> String {
+    let denom = unwrap_client_response(
+        RPC.vp()
+            .token()
+            .denomination(client, token, sub_prefix)
+            .await,
+    )
+    .unwrap_or_else(|| {
+        println!(
+            "No denomination found for token: {token}, defaulting to zero \
+             decimal places"
+        );
+        0.into()
+    });
+    DenominatedAmount { amount, denom }.to_string()
+}
+
+/// Make asset type corresponding to given address and epoch
+pub fn make_asset_type<T: std::fmt::Display>(
+    epoch: Epoch,
+    token: &Address,
+    sub_prefix: &Option<T>,
+    denom: MaspDenom,
+) -> AssetType {
+    // Typestamp the chosen token with the current epoch
+    let token_bytes = (
```

Comment:
Good call to create a `make_asset_type` function  as this prevents people from mixing up the tuple ordering. Another idea: maybe we can make a `MaspToken` or `DenominatedToken` struct to represent this quadruple. I say this, because this quadruple is also referred to frequently in the Merkle tree production code. I say a struct instead of a tuple because this structure now has 4 entries instead of 2, and hence field naming may make it easier to understand.